testsuite/tests/lib-obj/with_tag.ml

```diff
@@ -0,0 +1,31 @@
+(* TEST
+*)
+
+type t =
+| A of string * float
+| B of string * float
+
+let () =
+  assert (Obj.dup (Obj.repr (A ("hello", 10.))) = Obj.repr (A ("hello", 10.)));
+  assert (Obj.with_tag 1 (Obj.repr (A ("hello", 10.))) = Obj.repr (B ("hello", 10.)))
+
+let () =
+  assert (Obj.tag (Obj.with_tag 42 (Obj.repr [| |])) = 42)
```

Comment:
The empty array with tag 42 is going to be atom number 42, but (I think) atoms above 0 are currently not used by OCaml, so you are testing unused corners of the runtime. Maybe you would like to put an element in this array instead.

Reply:
I think it's the point of testing the empty atom, to avoid a regression that would have been present in an earlier version of the patch (where empty atoms were returned unchanged).